modules/salt/pages/large-scale-tuning.adoc

```diff
@@ -0,0 +1,524 @@
+| Notes                |
+| More information     | http://www.quartz-scheduler.org/documentation/2.4.0-SNAPSHOT/configuration.html
+|===
+
+
+
+=== `org.quartz.scheduler.idleWaitTime`
+
+[cols="1,1"]
+|===
+| Description          | Cycle time for Taskomatic.
+                         Decreasing this value lowers the latency of Taskomatic.
+| Tune when            | Client count is in the thousands.
+| Value default        | 5000 ms
+| Value recommendation | 1000-5000 ms
+| Location             | [path]``/etc/rhn/rhn.conf``
+| After changing       |
+| Notes                |
+| More information     | http://www.quartz-scheduler.org/documentation/2.4.0-SNAPSHOT/configuration.html
+|===
+
+
+
+=== `MinionActionExecutor.parallel_threads`
+
+[cols="1,1"]
+|===
+| Description          | Number of Taskomatic threads dedicated to sending commands to Salt clients as a result of Actions being executed.
+| Tune when            | Client count is in the thousands.
+| Value default        | 1
+| Value recommendation | 1-10
+| Location             | [path]``/etc/rhn/rhn.conf``
+| After changing       |
+| Notes                | Full parameter name is `taskomatic.com.redhat.rhn.taskomatic.task.MinionActionExecutor.parallel_threads`
+| More information     |
+|===
+
+
+
+=== `rhn-search.java.maxmemory`
+
+[cols="1,1"]
+|===
+| Description          | The maximum amount of memory that the `rhn-search` service can use.
+| Tune when            | Client count increases significantly, and `OutOfMemoryException` errors appear in `journalctl -u rhn-search`.
+| Value default        | 512 MiB
+| Value recommendation | 512-4096 MiB
+| Location             | [path]``/etc/rhn/rhn.conf``
+| After changing       | Check the Memory Usage section in this document.
+| Notes                |
+| More information     |
+|===
+
+
+
+=== `shared_buffers`
+
+[cols="1,1"]
+|===
+| Description          | The amount of memory reserved for PostgreSQL shared buffers, which contain caches of database tables and index data.
+| Tune when            | RAM changes
+| Value default        | 25% of total RAM
+| Value recommendation | 25-40% of total RAM
+| Location             | [path]``/var/lib/pgsql/data/postgresql.conf``
+| After changing       | Check the Memory Usage section in this document.
+| Notes                | The value is formatted as `<N>MB`, where `<N>` is the number of MiB to allocate.
+| More information     | https://www.postgresql.org/docs/10/runtime-config-resource.html#GUC-SHARED-BUFFERS
+|===
+
+
+
+=== `work_mem`
+
+[cols="1,1"]
+|===
+| Description          | The amount of memory allocated by PostgreSQL every time a connection needs to do a sort or hash operation.
+                         Every connection (as specified by `max_connections`) might make use of an amount of memory equal to a multiple of `work_mem`.
+| Tune when            | Individual query operations are too slow, and value is below 5 MB
+| Value default        |
+| Value recommendation | 2-20 MB
```

Comment:
```suggestion
| Value recommendation | 2-20 MiB
```

Reply:
Here the problem is all Postgres configuration variables are really set in MB, so I would not use MiB to save the reader a conversion.